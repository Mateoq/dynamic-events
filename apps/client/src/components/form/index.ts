export * from './Button';
export * from './SelectInput';
export * from './TextInput';
export * from './DatePicker';
export * from './CheckBox';
