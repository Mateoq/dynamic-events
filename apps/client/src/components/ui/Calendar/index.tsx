import React from 'react';

import { CalendarMonth } from './CalendarMonth';
import { CalendarWeek } from './CalendarWeek';
import { CalendarDay } from './CalendarDay';
import { CalendarNav } from './CalendarNav';
import { useCalendarStateStore } from '@/stores/calendar-state-store';
import { CalendarData, Direction, CalendarType } from '@/types';
import {
  getYearKey,
  getMonthKey,
  getWeekKey,
  getDayKey,
  getMonthFromState,
  getWeekFromState,
  getDayFromState
} from '@/utils/calendar';

export type CalendarProps = {
  type: CalendarType;
  data: CalendarData;
  onNav: (direction: Direction) => void;
  onCreateEvent: () => void;
  onChangeView: (type: CalendarType) => void;
  onClickEvent: (id: string) => void;
};

export const Calendar: React.FC<CalendarProps> = (props) => {
  const { type, data, onNav, onCreateEvent, onChangeView, onClickEvent } =
    props;
  const state = useCalendarStateStore((state) => state.calendarState);
  let currentComponent = (
    <div className="text-red-600 text-xl">Unknown Calendar</div>
  );

  console.log('DATA', data);

  if (state.year === 0 || state.month === 0 || state.day === 0) {
    return null;
  }

  switch (type) {
    case CalendarType.MOTH:
      const monthData = getMonthFromState(data, state);
      currentComponent = (
        <CalendarMonth data={monthData} onClickEvent={onClickEvent} />
      );
      break;
    case CalendarType.WEEK:
      const weekData = getWeekFromState(data, state);
      currentComponent = (
        <CalendarWeek data={weekData} onClickEvent={onClickEvent} />
      );
      break;
    case CalendarType.DAY:
      const dayData = getDayFromState(data, state);
      currentComponent = (
        <CalendarDay data={dayData} onClickEvent={onClickEvent} />
      );
      break;
  }

  return (
    <div className="w-full">
      <div className="w-full flex flex-col">
        <CalendarNav
          currentView={type}
          onNav={onNav}
          onCreateEvent={onCreateEvent}
          onChangeView={onChangeView}
        />
        {currentComponent}
      </div>
    </div>
  );
};
