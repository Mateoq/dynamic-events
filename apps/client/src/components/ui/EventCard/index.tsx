import React from 'react';

import { Event } from '@/types';

export type EventCardProps = {
  data: Event;
  onClick?: (id: string) => void;
};

export const EventCard: React.FC<EventCardProps> = (props) => {
  const { data, onClick } = props;
  return (
    <button
      className="flex-1 px-1 overflow-hidden"
      style={{ background: data.color }}
      onClick={() => onClick?.(data.id)}
    >
      <p className="text-white text-sm truncate">{data.title}</p>
    </button>
  );
};
