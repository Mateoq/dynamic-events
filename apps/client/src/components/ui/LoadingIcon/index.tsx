import React from 'react';
import { motion } from 'framer-motion';
import { FaSpinner } from 'react-icons/fa6';

export type LoadingIconProps = {
  color?: string;
};

export const LoadingIcon: React.FC<LoadingIconProps> = (props) => {
  const { color = 'white' } = props;
  return (
    <motion.div
      className="w-4 h-4 round-full"
      animate={{ rotate: [0, 360] }}
      transition={{ duration: 1, ease: 'linear', repeat: Infinity }}
    >
      <FaSpinner color={color} />
    </motion.div>
  );
};
