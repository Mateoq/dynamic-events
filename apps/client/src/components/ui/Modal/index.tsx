import React from 'react';
import { AnimatePresence, motion } from 'framer-motion';
import { FaXmark } from 'react-icons/fa6';

export type ModalProps = {
  isOpen?: boolean;
  title: string;
  onClose: () => void;
};

export const Modal: React.FC<React.PropsWithChildren<ModalProps>> = (props) => {
  const { isOpen, title, children, onClose } = props;

  return (
    <AnimatePresence>
      {isOpen && (
        <motion.div
          initial={{ opacity: 0 }}
          animate={{ opacity: 1 }}
          exit={{ opacity: 0 }}
          className="w-screen h-screen fixed top-0 left-0 flex items-center justify-center z-50 overflow-visible"
        >
          <div className="w-[500px] h-fit bg-white rounded shadow-2xl relative flex flex-col">
            <div className="flex justify-between h-fit py-2 px-2 bg-gray-500 rounded-t">
              <h3 className="text-lg text-gray-50 font-medium">{title}</h3>
              <button
                onClick={onClose}
                aria-label="Close Modal"
                className="h-6 w-6 transition-opacity hover:opacity-80 active:opacity-100"
              >
                <FaXmark size="100%" className="text-gray-50" />
              </button>
            </div>
            <div className="px-3 py-5">{children}</div>
          </div>
        </motion.div>
      )}
    </AnimatePresence>
  );
};
