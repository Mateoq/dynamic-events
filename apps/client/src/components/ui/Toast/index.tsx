import React from 'react';
import { AnimatePresence, motion } from 'framer-motion';
import clsx from 'clsx';

import { ToastType } from '@/types';

export type ToastProps = {
  isOpen?: boolean;
  type: ToastType;
  text: string;
};

export const Toast: React.FC<ToastProps> = (props) => {
  const { isOpen, type, text } = props;
  const isSuccess = type === ToastType.SUCCESS;
  const isError = type === ToastType.ERROR;

  return (
    <AnimatePresence>
      {isOpen && (
        <motion.div
          initial={{ x: '100%', opacity: 0 }}
          animate={{ x: 0, opacity: 1 }}
          exit={{ x: '100%', opacity: 0 }}
          transition={{ duration: 0.2 }}
          className={clsx(
            'w-52 min-h-24 rounded filter drop-shadow-lg flex items-center justify-center p-5 z-[100] fixed bottom-5 right-5',
            { 'bg-gray-900': isSuccess, 'bg-red-600': isError }
          )}
        >
          <span className="text-md text-gray-100">{text}</span>
        </motion.div>
      )}
    </AnimatePresence>
  );
};
