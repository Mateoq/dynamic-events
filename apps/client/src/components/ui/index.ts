export * from './Calendar';
export * from './LoadingIcon';
export * from './Modal';
export * from './Toast';
export * from './EventCard';
