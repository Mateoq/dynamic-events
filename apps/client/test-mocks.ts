import { CalendarState, Event, TransactionResultType } from '@/types';
import { findDayWeek, initCalendarData } from '@/utils/calendar';
import { newTransactionResult } from '@/helpers';

export const emptySuccessTransactionResult = newTransactionResult(
  null,
  TransactionResultType.SUCCESS,
  true
);

export const singleEvent: Event = {
  id: 'test-id',
  title: 'Test Event',
  description: 'Test description',
  city: 'New York',
  startYear: 2024,
  startMonth: 9,
  startDate: 17,
  endYear: 2024,
  endMonth: 9,
  endDate: 20,
  time: 13,
  fullDay: false,
  color: '#FF00FF'
};

export const initialCalendarData = initCalendarData(new Date(2024, 9, 17));

export function createInitialCalendarState() {
  const date = new Date();
  const year = date.getFullYear();
  const month = date.getMonth();
  const week = findDayWeek(initialCalendarData, date);
  const day = date.getDate();
  const initialState: CalendarState = {
    year,
    month,
    week,
    day
  };

  return initialState;
}

const events = {
  events: {
    'caeeb5f1-6788-4b1f-bbf9-ee27b19dc84f': {
      id: 'caeeb5f1-6788-4b1f-bbf9-ee27b19dc84f',
      title: 'Test',
      description: 'asfasf',
      city: 'New York',
      startYear: 2024,
      startMonth: 9,
      startDate: 14,
      endYear: 2024,
      endMonth: 9,
      endDate: 16,
      time: 14,
      fullDay: false,
      color: '#c6cd65'
    },
    '9ca87c5d-ac5d-4cca-9510-fefffbb3a801': {
      id: '9ca87c5d-ac5d-4cca-9510-fefffbb3a801',
      title: 'ANother',
      description: 'gasasf',
      city: 'New York',
      startYear: 2024,
      startMonth: 9,
      startDate: 14,
      endYear: 2024,
      endMonth: 9,
      endDate: 25,
      time: 17,
      fullDay: false,
      color: '#4c94c8'
    },
    'df42f359-04ee-4b46-acac-aa55b2b76fa8': {
      id: 'df42f359-04ee-4b46-acac-aa55b2b76fa8',
      title: 'Test 2',
      description: 'asfasf',
      city: 'New York',
      startYear: 2024,
      startMonth: 9,
      startDate: 14,
      endYear: 2024,
      endMonth: 9,
      endDate: 14,
      time: 14,
      fullDay: false,
      color: '#543bc4'
    }
  },
  calendarEvents: {
    '2024': {
      '9': {
        '14': {
          '14': [
            'caeeb5f1-6788-4b1f-bbf9-ee27b19dc84f',
            'df42f359-04ee-4b46-acac-aa55b2b76fa8'
          ],
          '17': ['9ca87c5d-ac5d-4cca-9510-fefffbb3a801']
        },
        '15': {
          '14': ['caeeb5f1-6788-4b1f-bbf9-ee27b19dc84f'],
          '17': ['9ca87c5d-ac5d-4cca-9510-fefffbb3a801']
        },
        '16': {
          '14': ['caeeb5f1-6788-4b1f-bbf9-ee27b19dc84f'],
          '17': ['9ca87c5d-ac5d-4cca-9510-fefffbb3a801']
        },
        '17': {
          '17': ['9ca87c5d-ac5d-4cca-9510-fefffbb3a801']
        },
        '18': {
          '17': ['9ca87c5d-ac5d-4cca-9510-fefffbb3a801']
        },
        '19': {
          '17': ['9ca87c5d-ac5d-4cca-9510-fefffbb3a801']
        },
        '20': {
          '17': ['9ca87c5d-ac5d-4cca-9510-fefffbb3a801']
        },
        '21': {
          '17': ['9ca87c5d-ac5d-4cca-9510-fefffbb3a801']
        },
        '22': {
          '17': ['9ca87c5d-ac5d-4cca-9510-fefffbb3a801']
        },
        '23': {
          '17': ['9ca87c5d-ac5d-4cca-9510-fefffbb3a801']
        },
        '24': {
          '17': ['9ca87c5d-ac5d-4cca-9510-fefffbb3a801']
        },
        '25': {
          '17': ['9ca87c5d-ac5d-4cca-9510-fefffbb3a801']
        }
      }
    }
  }
};

export function getEvents() {
  return events;
}
