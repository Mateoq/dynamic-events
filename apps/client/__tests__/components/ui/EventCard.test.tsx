import '@testing-library/jest-dom';
import { render, screen } from '@testing-library/react';

import { EventCard } from '@/components/ui';
import { singleEvent } from '../../../test-mocks';

describe('EventCard', () => {
  it('should match snapshot', async () => {
    const { container } = render(<EventCard data={singleEvent} />);
    const title = await screen.findByText(singleEvent.title);

    expect(title).toBeInTheDocument();
    expect(container).toMatchSnapshot();
  });
});
