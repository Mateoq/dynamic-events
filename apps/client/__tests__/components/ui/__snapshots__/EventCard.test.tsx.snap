// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`EventCard should match snapshot 1`] = `
<div>
  <button
    class="flex-1 px-1 overflow-hidden"
    style="background: rgb(255, 0, 255);"
  >
    <p
      class="text-white text-sm truncate"
    >
      Test Event
    </p>
  </button>
</div>
`;
